src/canvas-force-graph.js

```diff
@@ -154,60 +178,72 @@ export default Kapsule({
 
         const visibleLinks = state.graphData.links.filter(getVisibility);
 
-        if (state.linkCanvasObject) {
-          // Custom link paints
-          visibleLinks.forEach(link => state.linkCanvasObject(link, state.ctx, state.globalScale));
-          return;
-        }
-
         // Bundle strokes per unique color/width for performance optimization
-        const linksPerColor = indexBy(visibleLinks, [getColor, getWidth]);
+        const linksPerColor = indexBy(visibleLinks, [getColor, getWidth, getLinkCanvasObjectMode]);
```

Comment:
Because, unless I understood it wrong (quite probably), it is after the links iteration that all the links are drawn at once by this piece of code.

```Javascript
ctx.strokeStyle = lineColor;
ctx.lineWidth = lineWidth;
ctx.stroke();
```

Now, if we do that _regardless_ of some links having their rendering replaces by the `linkCanvasObject` function, the default drawing still applies. Look at this screenshot of the `dash-odd-links` example:

![Captura de pantalla 2019-05-21 a las 20 29 40](https://user-images.githubusercontent.com/1730892/58121235-35c18100-7c07-11e9-858c-3b860abd6196.png)

That's why I struggled with this method. 😅 

Am I missing something?

Reply:
I think I've found a nicer way to do this that doesn't require the coupled iteration. If you give me PR edit permissions (https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) I'll show you the changes.

Reply:
The _Allow allow edits from maintainer_ checkbox is checked. Please, go ahead!